### PR TITLE
More steps

### PR DESCRIPTION
**Pull request by galderz**
This is a PR to start getting some feedback on the updated workshop steps.

For now, I've just added installation steps that includes Virtual Box set up and individual IDE instructions for importing the workshop.

I've also removed the second exercise since we won't deliver this for Codemotion Madrid.

Next I'll start going through each exercise and I'll add context information on the goal of the exercise, a data flow diagram to go along with the existing diagram and theory required to do that exercise. Those parts that require explaining Vert.x theory I'll leave them empty for Thomas to fill in.

### CODE REVIEW

**Comment by galderz** [comment]
I've added one commit with the first bit of theory, which explains what is OpenShift. I was thinking that to give consistency to the steps, we could use green boxes for anything theory related. This will make it easy to spot the theory bits.

**Comment by galderz** [comment]
I've added more details, screenshots...etc to the first exercise. I'll address the rest of exercises next.

@tsegismont I've left some blank spaces at the beginning of first exercise for you to explain what is vert.x, what is its role in the workshop...etc.

I'm thinking of leaving adding data flow diagrams to the very last. So, I'll add more details, screenshots and theory for the rest of exercises next.

**Comment by galderz** [comment]
I've pushed updates to the data injection and delayed train dashboard exercises.

On Monday I will complete the remaining exercise and will update the code based on the RxJava2 changes @tsegismont did on master.

After that, I'll create a separate issue for adding the data flow diagrams, which I'll do later in the week.

File: workshop-steps/workshop.html [hunk 1]
```diff
@@ -49,258 +49,474 @@
...
+  <br/><br/>
+  <pre><code>> cd streaming-data-workshop
+> git fetch origin
+> git pull</code></pre>
+  <br/>
+  <img src="img/update-workshop.png" class="img-rounded" height="75%" width="75%">
+  <br/><br/>
+
+  <div class="alert alert-danger" role="alert">
+    If you're running this workshop <strong>completely offline</strong>, you need to adjust <code>/etc/hosts</code> file inside the virtual machine so that host names can be resolved locally:
```
Comment (tsegismont) [inline workshop-steps/workshop.html]: How about telling them to copy/paste this into the terminal instead?

```
sudo sh -c 'cat >> /etc/hosts << EOF
127.0.0.1        datagrid-visualizer-myproject.127.0.0.1.nip.io
127.0.0.1        delayed-listener-myproject.127.0.0.1.nip.io
127.0.0.1        delayed-trains-myproject.127.0.0.1.nip.io
127.0.0.1        workshop-main-myproject.127.0.0.1.nip.io
EOF'
```

File: workshop-steps/workshop.html [hunk 2]
```diff
@@ -49,258 +49,474 @@
 </div>
 
 <div class="container">
-  <h1>Playground</h1>
+  <h1>Installation Steps</h1>
+  To do this workshop, it's necessary to have some components installed in the local machine.
+  Depending on your environment, you can find detailed installation instructions below:
```
Comment (tsegismont) [inline workshop-steps/workshop.html:54]: Can you add two links to the relevant sections here? Like:

```html
<ul>
  <li><a href="#_restricted-internet-environment">Restricted Internet Environment</a</li>
  <li><a href="#_unrestricted-internert-environment">Unrestricted Internet Environment</a</li>
</ul>
```

**Review by tsegismont — APPROVED** [review]
Very nice @galderz ! Just a couple of concerns

**Comment by galderz** [comment]
I've just pushed changes for the last exercise. 

@tsegismont I'll address your comments in a sec.

The final thing remaining is updating steps with the correct solutions based on @tsegismont's RxJava2 changes. The data flows, as mentioned above, will be done later this week.

File: workshop-steps/workshop.html (hunk 2 above)
Comment (galderz) [reply workshop-steps/workshop.html:54]: Ok, I'll add those inside installation steps, should be quick.

I do think we need a full table of contents at the top though as well as this, what do you think?

Comment (tsegismont) [reply workshop-steps/workshop.html:54]: If eventually we move the content to Asciidoc, then the TOC will come for free. In the meantime, I'm not really keen on maintaining one manually :-)

**Comment by tsegismont** [comment]
@galderz where did you push the commits? I can't see them here

**Comment by galderz** [comment]
@tsegismont I had not fully pushed the last exercise. It should be there now.

**Comment by galderz** [comment]
I've fixed steps for RxJava2.

This PR is ready as far as I'm concerned, with only data flow diagrams to be done later this week (I'll create an issue for that)

@karesti What do you think?

**Comment by galderz** [comment]
You can quickly preview the steps via [this link](http://htmlpreview.github.io/?https://github.com/infinispan-demos/streaming-data-workshop/blob/more-steps/workshop-steps/workshop.html).

I've updated root README to point to workshop steps directly.

**Comment by tsegismont** [comment]
@galderz Exercice2 TODO 1 I see this:

`Flowable> pairFlowable = fileFlowable.map(StationsInjector::toEntry);`

**Comment by tsegismont** [comment]
And the bottom of the page looks like content from a JS file

**Review by tsegismont — CHANGES_REQUESTED** [review]
@galderz something went wrong with your last updates, the page is broken (see my other comments)

**Comment by galderz** [comment]
> Re: And the bottom of the page looks like content from a JS file

That's just the htmlpreview thing... just run `open workshop-steps/workshop.html` from local.

**Comment by galderz** [comment]
> @galderz Exercice2 TODO 1 I see this:

Those are problems with java generics in html. Don't know how to solve them

**Comment by karesti** [comment]
@galderz I've looked to the PR and globally seems good to me. I don't request any particular change

**Review by tsegismont — APPROVED** [review]

**Comment by tsegismont** [comment]
@galderz I pushed a commit to fix the 2 samples with `Flowable` and generics

**Comment by galderz** [comment]
Thx @tsegismont, let's integrate it then :)

**Pull request merged**